src/utils.ts

```diff
@@ -15,7 +15,15 @@ export interface ExecResult {
   error?: cp.ExecFileException;
 }
 
-export async function exec(command: string, args: string[], options: cp.ExecFileOptions = { shell: true }) {
+export async function exec(
+  command: string,
+  args: string[],
+  extraEnv: { [key: string]: string } | undefined = undefined,
+  options: cp.ExecFileOptions = { shell: true },
+) {
+  if (extraEnv) {
+    options.env = { ...(options.env ?? process.env), ...extraEnv };
```

Comment:
Maybe I don't understand the node API here, if we pass an environment object, does it not also inherit the parent environment?

Reply:
I'm not 100% sure. The documentation says, about the `env` option:

> `env`: Environment key-value pairs. Default: `process.env`.

So because the default is `process.env` and not `{}`, I'd guess that `options.env` replaces the parent environment. (That also makes sense because otherwise, you couldn't delete environment variables.)